🧪 Pin codecov-cli deptree used in AZP

webknjaz opened the pull request:
##### SUMMARY

`codecov-cli == 11.0.3` allows `click == 8.3.0` in its deps but the latter causes commit auto-discovery breakage in the former. With https://github.com/getsentry/prevent-cli/pull/95, `codecov-cli == 11.2.3` excludes this version so this patch updates the requirement to that.

To prevent history from repeating itself, this change also makes use of a pip constraint file that pins the entire dependency tree to concrete versions.

Example failure in our CI from today's morning: https://dev.azure.com/ansible/ansible/_build/results?buildId=157938&view=logs&j=d7668ad9-d7bb-5ae4-c14f-5061b89e467d&t=7f884d87-6a36-516f-9067-af4cf77c020d&l=71

##### ISSUE TYPE

- Test Pull Request

webknjaz committed:
🧪 Pin `codecov-cli` deptree used in AZP

This excludes the incompatible version of Click v8.3.0 from the
dependency resolution.

Refs:
* https://github.com/getsentry/prevent-cli/pull/95
* https://github.com/pallets/click/issues/3066

webknjaz committed:
🧪 Drop manual Click exclusion

ci_coverage

webknjaz committed:
💅 Make `pip-compile` inline source comments

ci_coverage
ci_complete

mattclay reviewed (approved):
Looks good. Just be sure to take a look at the coverage results on codecov.io after tests finish to make sure they're looking reasonably close to what we usually see.

webknjaz committed:
󰨮Compute absolute requirement/constraint paths

When they are relative to CWD, it's unreliable as the script does not
know where it'd be invoked from.

ci_coverage
ci_complete
